operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java

```diff
@@ -86,6 +87,9 @@ void configureHostname() {
                 .mapOption("hostname-admin-url", HostnameSpec::getAdminUrl)
                 .mapOption("hostname-strict", HostnameSpec::isStrict)
                 .mapOption("hostname-strict-backchannel", HostnameSpec::isStrictBackchannel);
+        optionMapper(Function.identity())
+                .mapOption("proxy", kc -> (!CRDUtils.isTlsConfigured(kc)
+                        && KeycloakIngressDependentResource.isOpenshiftDefaultIngressEnabled(kc)) ? "edge" : null);
```

Comment:
@shawkins Maybe I'm missing a broader context, but why do we need to set `proxy=edge` only when Openshift Ingress is enabled? We could do it even for the default Ingress when it's enabled and TLS is not configured, right? Could you elaborate on it a little bit for me or point me to a related discussion? 

Thanks!

Reply:
> We could do it even for the default Ingress when it's enabled and TLS is not configured, right?

That's what I'm trying to get answered here https://github.com/keycloak/keycloak/issues/23716#issuecomment-1752883734